File: it/src/test/java/org/zalando/nakadi/client/examples/java/EventCreationExample.java
package org.zalando.nakadi.client.examples.java;

import java.util.ArrayList;
import java.util.List;

import org.zalando.nakadi.client.java.Client;
import org.zalando.nakadi.client.java.model.Event;
import org.zalando.nakadi.client.java.model.EventEnrichmentStrategy;
import org.zalando.nakadi.client.java.model.EventType;
import org.zalando.nakadi.client.java.model.EventTypeCategory;
import org.zalando.nakadi.client.java.model.EventTypeSchema;
import org.zalando.nakadi.client.java.model.EventValidationStrategy;
import org.zalando.nakadi.client.java.model.PartitionStrategy;
import org.zalando.nakadi.client.java.model.SchemaType;
import org.zalando.nakadi.client.scala.ClientFactory;
import org.zalando.nakadi.client.utils.ClientBuilder;

public class EventCreationExample {

	public class MeetingsEvent implements Event {
		private final String date;
		private final String topic;

		public MeetingsEvent(String date, String topic) {
			this.date = date;
			this.topic = topic;
		}

		public String getDate() {
			return date;
		}

		public String getTopic() {
			return topic;
		}

	}

	public Client createClient() {
		return new ClientBuilder()//
				.withHost("nakadi-sandbox.aruha-test.zalan.do")//
				.withSecuredConnection(true) // s
				.withVerifiedSslCertificate(false) // s
				.withTokenProvider4Java(() -> ClientFactory.getToken())//
				.buildJavaClient();
	}

	public EventTypeSchema createEventTypeSchema(String schema) {
		return new EventTypeSchema(SchemaType.JSON, schema);
	}

	public EventType createEventType(String name,EventTypeSchema eventTypeSchema) {
		String owner = "team-laas";
		EventTypeCategory category = EventTypeCategory.UNDEFINED;
		EventValidationStrategy validationStrategies = EventValidationStrategy.NONE;
		EventEnrichmentStrategy enrichmentStrategies = null;
		PartitionStrategy partitionStrategy = PartitionStrategy.RANDOM;
		 List<String> paritionKeyFields  = new ArrayList("date", "topic");
		 
		return new EventType(name,// eventTypeName
				owner,// owner
				category,// category
				validationStrategies,// validationStrategies
				enrichmentStrategies,// enrichmentStrategies
				partitionStrategy,// partitionStrategy
				eventTypeSchema,// eventTypeSchema
				null,// dataKeyFields
				null,// partitionKeyFields
				null);// statistics

	}

	public static void main(String[] args) {
		EventCreationExample example = new EventCreationExample();
		// 1. Create client
		final Client client = example.createClient();

		// 2. Create a simple Meeting instance
		EventCreationExample.MeetingsEvent meeting = example.new MeetingsEvent(
				"2016-04-28T13:28:15+00:00", "Hackthon");

		// Create the EventType
		String schema = " { \"properties\":\" { \"date\": { \"type\": \"string\" }, \"topic\": { \"type\": \"string\"} } }";

		EventTypeSchema eventSchame = example.createEventTypeSchema(schema);

	}
}

File: it/src/test/scala/org/zalando/nakadi/client/examples/scala/EventCreationExample.scala
package org.zalando.nakadi.client.examples.scala

import org.zalando.nakadi.client.scala.Client
import org.zalando.nakadi.client.scala.model.Event
import org.zalando.nakadi.client.scala.model.EventType
import org.zalando.nakadi.client.scala.model.EventTypeCategory
import org.zalando.nakadi.client.scala.model.EventTypeSchema
import org.zalando.nakadi.client.scala.model.ScalaJacksonJsonMarshaller
import org.zalando.nakadi.client.scala.model.PartitionStrategy
import org.zalando.nakadi.client.scala.model.SchemaType
import org.zalando.nakadi.client.utils.ClientBuilder
import scala.concurrent.Await
import scala.concurrent.duration.DurationInt
import scala.collection.mutable.ListBuffer
import org.zalando.nakadi.client.scala.ClientFactory
import org.zalando.nakadi.client.scala.model.EventEnrichmentStrategy
import scala.concurrent.ExecutionContext.Implicits.global

object EventCreationExample extends App {

  // 1. Create the client

  import org.zalando.nakadi.client.utils.ClientBuilder
  import org.zalando.nakadi.client.scala.Client
  val client: Client = ClientFactory.buildScalaClient()

  // 2. Create a simple Meeting Event class
  case class MeetingsEvent(date: String, topic: String) extends Event

  //We define the Json representation of our Meeting Event.
  val schema: String = """
    { 
      'properties': 
      { 
        'date': { 'type': 'string' }, 
        'topic': { 'type': 'string'}  
      } 
    }""".replaceAll("'", "\"")

  // 3. Create the EventType, 
  // We need to create an eventType(a topic), where listeners 
  // can subscribe to, to get our published events.
  val eventTypeSchema = new EventTypeSchema(SchemaType.JSON, schema)

  //First the eventType name, wich will be part of the URL: https://nakadi.test.io/event-types/{eventTypeName}
  //See the API for more information on the EventType model
  //https://github.com/zalando/nakadi/blob/nakadi-jvm/api/nakadi-event-bus-api.yaml#L1240
  //  val eventTypeName = "Example-unique-million-messages"
  val eventTypeName = "idakan-test-event"
  //  val eventTypeName = "Example-unique-hundred-messages-3"

  val owner = "team-laas"
  val category = EventTypeCategory.UNDEFINED // We want just to pass data without through Nakadi, simple schema-validation is enough!
  val enrichmentStrategies = Nil //Seq(EventEnrichmentStrategy.METADATA)
  val partitionStrategy = Some(PartitionStrategy.RANDOM)
  val dataKeyFields = Nil
  val paritionKeyFields = List("date", "topic")

  val eventType = new EventType(eventTypeName,
    owner,
    category,
    enrichmentStrategies,
    partitionStrategy,
    eventTypeSchema,
    dataKeyFields,
    paritionKeyFields,
    None)

  //You need to import the default Serializer if you don't sepecify your own!
  import ScalaJacksonJsonMarshaller._

  client.createEventType(eventType)
  Thread.sleep(10000)
  // 4. Publish the EventType
  //  System.exit(0)
  var counter = 0
  for (n <- 1 to 50) {
    val event = new MeetingsEvent("" + System.currentTimeMillis(), "Hackaton")
    var events = ListBuffer[MeetingsEvent]()
    for (a <- 1 to 10) {
      counter += 1
      events += MeetingsEvent("" + System.currentTimeMillis(),
        "Hackaton" + counter)
    }
    Await.result(client.publishEvents(eventTypeName, events), 10.seconds)
  }
  client.stop()

}
